vimoutliner/scripts/votl_maketags.py

```diff
@@ -0,0 +1,119 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+#   Based on votl_maketags.py
+#   Copyright (C) 2001-2003, 2011 by Steve Litt (slitt@troubleshooters.com)
+#   This script is
+#   Copyright (C) 2015 Matěj Cepl <mcepl@cepl.eu>
+#
+#   This program is free software; you can redistribute it and/or modify
+#   it under the terms of the GNU General Public License as published by
+#   the Free Software Foundation; either version 2 of the License, or
+#   (at your option) any later version.
+#
+#   This program is distributed in the hope that it will be useful,
+#   but WITHOUT ANY WARRANTY; without even the implied warranty of
+#   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+#   GNU General Public License for more details.
+#
+#   You should have received a copy of the GNU General Public License
+#   along with this program; if not, see <http://www.gnu.org/licenses/>.
+from __future__ import print_function
+import os.path
+import os
+import re
+import sys
+import subprocess
+import shutil
+import tempfile
+
+TAGFILENAME = os.path.expanduser("~/.vim/vimoutliner/vo_tags.tag")
+
+
+def makeTagFileStartingAt(args):
+    if len(args) != 1:
+        print("""
+        Usage is:
+            otltags topLevelOutlineFileName""")
+        sys.exit(1)
+
+    absoluteFileName = os.path.abspath(args[0])
+
+    processedFiles = {}
+    recordFileAsProcessed(absoluteFileName, processedFiles)
+    os.unlink(TAGFILENAME)
+    process1Outline(absoluteFileName, processedFiles)
+    sortAndDeleteDupsFromTagFile(absoluteFileName)
+
+
+def sortAndDeleteDupsFromTagFile(infile):
```

Comment:
True, I totally agree that it is best to use `sort` in the case of large files, but this is the tag file that is being sorted. Having that would require having millions of tags. That would be premature optimisation.

The problem with calling out to programs is that they aren't guaranteed to be everywhere. `sort` does not exist on all systems with the same flags (e.g., Windows).